pkg/api/v1alpha1/cluster_defaults.go

```diff
@@ -51,3 +52,20 @@ func setWorkerNodeGroupDefaults(cluster *Cluster) error {
 	}
 	return nil
 }
+
+func setCNIConfigDefault(cluster *Cluster) error {
```

Comment:
test for this method?

Reply:
I thought [these changes](https://github.com/aws/eks-anywhere/pull/1434/files#diff-c05e092d1a8232f079da26d00f68592a14a768e67bde0423768134e57b3815e0) would cover it, but I just realized in that I need to update the `in` cluster to use the deprecated field, will do that

Reply:
I think we should have a test that asserts that the structs are being set (they are not nil)
So add two testcases to the existing test

Reply:
the changes in the existing testcase do assert that they are not nil, since CNIConfig and CiliumConfig in the `wantCluster` are initialized. So I think that is being tested. But still I separated out the test cases, so the original test for worker node group name does just that, and takes in a non-nil CNIConfig. and the new test case takes in nil CNIConfig and the `wantCluster` contains non-nil CNIConfig & CiliumConfig